File: lib/utils/polish_holidays.dart
class PolishHolidays {
  static bool isHoliday(DateTime date) {
    final month = date.month;
    final day = date.day;

    // Fixed holidays
    const fixedHolidays = [
      (1, 1), // New Year's Day
      (1, 6), // Epiphany
      (5, 1), // Labour Day
      (5, 3), // Constitution Day
      (8, 15), // Assumption of Mary
      (11, 1), // All Saints' Day
      (11, 11), // Independence Day
      (12, 25), // Christmas
      (12, 26), // Second Day of Christmas
    ];

    for (final holiday in fixedHolidays) {
      if (month == holiday.$1 && day == holiday.$2) {
        return true;
      }
    }

    // Easter-based holidays (approximate for demo - you can add precise Easter calculation)
    // For now, we'll skip Easter calculations, but in production you'd want to add:
    // - Easter Sunday
    // - Easter Monday
    // - Corpus Christi (60 days after Easter)

    return false;
  }

  static String? getHolidayName(DateTime date) {
    final month = date.month;
    final day = date.day;

    const holidays = {
      (1, 1): 'Nowy Rok',
      (1, 6): 'Święto Trzech Króli',
      (5, 1): 'Święto Pracy',
      (5, 3): 'Święto Konstytucji 3 Maja',
      (8, 15): 'Wniebowzięcie NMP',
      (11, 1): 'Wszystkich Świętych',
      (11, 11): 'Święto Niepodległości',
      (12, 25): 'Boże Narodzenie',
      (12, 26): 'Drugi Dzień Bożego Narodzenia',
    };

    return holidays[(month, day)];
  }
}

File: work-tracker-react/src/utils/polishHolidays.ts
import { format } from 'date-fns';

const POLISH_HOLIDAYS: Record<string, string> = {
    '01-01': 'Nowy Rok',
    '01-06': 'Święto Trzech Króli',
    '05-01': 'Święto Pracy',
    '05-03': 'Święto Konstytucji 3 Maja',
    '08-15': 'Wniebowzięcie NMP',
    '11-01': 'Wszystkich Świętych',
    '11-11': 'Święto Niepodległości',
    '12-25': 'Boże Narodzenie',
    '12-26': 'Drugi Dzień Bożego Narodzenia'
};

export function isHoliday(date: Date): boolean {
    const key = format(date, 'MM-dd');
    return key in POLISH_HOLIDAYS;
}

export function getHolidayName(date: Date): string | null {
    const key = format(date, 'MM-dd');
    return POLISH_HOLIDAYS[key] || null;
}

export function isWeekend(date: Date): boolean {
    const day = date.getDay();
    return day === 0 || day === 6; // Sunday or Saturday
}
